File: apps/query-burst-msa/catalog-service/src/main/java/com/booster/queryburstmsa/catalog/web/dto/CategoryResponse.java
package com.booster.queryburstmsa.catalog.web.dto;

import com.booster.queryburstmsa.catalog.domain.entity.CategoryEntity;

public record CategoryResponse(
        Long id,
        String name,
        Long parentId,
        int depth
) {
    public static CategoryResponse from(CategoryEntity entity) {
        return new CategoryResponse(
                entity.getId(),
                entity.getName(),
                entity.getParentId(),
                entity.getDepth()
        );
    }
}

File: apps/kotlin/shopping-service/src/main/kotlin/com/booster/kotlin/shoppingservice/catalog/web/dto/response/CategoryResponse.kt
package com.booster.kotlin.shoppingservice.catalog.web.dto.response

import com.booster.kotlin.shoppingservice.catalog.domain.Category

data class CategoryResponse(
    val id: Long,
    val name: String,
    val children: List<CategoryResponse>,

) {
    companion object {
        fun from(category: Category): CategoryResponse = CategoryResponse(
            id = category.id,
            name = category.name,
            children = category.children.map { from(it) },
        )
    }
}
